### ISSUE TITLE
Feature request: Implement a method for selecting a Range of Tracks

### ISSUE THREAD

**Issue by oluc** [BasicAirData/GPSLogger#120]
is there a way to export all tracks at once?

**Comment by GrazianoCapelli** [comment]
So far, if you want to export all the Tracks at once, you must select (one by one) all the Tracks of the Tracklist and then click the "Export" Button. You can export also the current Track during the recording.

If I understand well, you would like the possibility to select all the tracks in one click, avoiding the "one by one" selection.
This case, If you agree, We could keep this Issue open as a Feature Request.

When the Feature Request is listed here, it could be chosen in the future by a Developer and it could be implemented.
We should start with a mock-up of the User Interface, and this would be the place for ideas, mock-ups, and discussion.

**Comment by oluc** [comment]
Yes, I'd like to export all tracks "in one click".

To add a bit of context to my personal situation, after issue #119, I wanted to keep a backup of my tracks.
- `$ adb backup` pulls the database. The data is here, but not very usable for other purposes
- selecting all tracks one by one is not really an option, as I have over 600 tracks


Since the **[export]** button already exports all selected tracks, maybe all that's missing is a **[select all]** button, that appears in the toolbar after you selected one.

1. select one (long tap)
--> toolbar appears
2. tap **[select all]** button
3. tap **[export]** button
==> all exported

This would work.


Additionally, 
in most apps with lists, you can select multiple items by marking the first and last item. It seems this is not working in GPSLogger. This could be handy to select a range of tracks (eg. "all tracks from last year")

1. long tap an item
--> item is selected
2. scroll down
3. long tap another item
==> all items in between are selected


This would also be usefull to **[delete]** a wide range of tracks.

**Comment by GrazianoCapelli** [comment]
Thanks for the useful and inspiring suggestions you posted.

I'm trying to pick from your ideas in order to find a way to implement the range selection without adding buttons and without changing the user experience.
Following these requirements, another possibility could be the following one:
- A short click selects/deselects a track
- A long click select/deselect a range of tracks (from the previous click to the current)

This way the users could select a range of tracks by following these steps:
1) select the first track (with a short click)
2) scroll down
3) long click on the last track to select all the range

**Comment by GrazianoCapelli** [comment]
The feature has been implemented, and will be added into the next App's Update v2.3.1.
The implementation is the following:
- A short click selects/deselects a track (behaviour not changed)
- A long click selects/deselects a range of tracks. The range is defined from the previous click to the current one (extents included).

When the user perform a long Click, the Tracklist check the last selection (the start point of the range) and operate this way:
If the previous track clicked is SELECTED, it selects all the Tracks of the Range
If the previous track clicked is DESELECTED, it deselects all the Tracks of the Range

I've tried this logic and I found it quite intuitive. I hope it will be good for the users too.
I think we could implement it with this Logic.
We can choose to change the Behaviour anytime in case of bad Feedbacks or better Ideas.

_Time for suggestions..._
